operators/pkg/info/info.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package info
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const UUIDCfgMapName = "elastic-operator-uid"
+
+var (
+	// Variables that must be set during the manager initialization by calling `Setup(ns, cfg)`.
+	operatorNamespace string
+	k8sDistribution   string
+
+	// Operator UUID set through a config map reconciliation.
+	operatorUUID types.UID
+)
+
+// Info contains versioning information.
+type Info struct {
+	UUID         types.UID `json:"uuid"`
+	Version      Version   `json:"version"`
+	Namespace    string    `json:"namespace"`
+	Distribution string    `json:"distribution"`
+}
+
+// Version contains number and build metadata information.
+type Version struct {
+	Number        string `json:"number"`
+	BuildHash     string `json:"build_hash"`
```

Comment:
Why not store version in its own field, and `Build` in a different struct?

Reply:
I have shamefully copied Elasticsearch:
```bash
{
  "number": "7.1.1",
  "build_flavor": "default",
  "build_type": "docker",
  "build_hash": "7a013de",
  "build_date": "2019-05-23T14:04:00.380842Z",
  "build_snapshot": false,
  "lucene_version": "8.0.0",
  "minimum_wire_compatibility_version": "6.8.0",
  "minimum_index_compatibility_version": "6.0.0-beta1"
}
```

Reply:
Ack :)